src/bounds.rs

```diff
@@ -0,0 +1,224 @@
+                let bounds = Bounds::Commits {
+                    start: date_to_sha(&start)?,
+                    end: date_to_sha(&end)?,
+                };
+                return Ok(bounds);
+            }
+        }
+        Ok(bounds)
+    }
+}
+
+/// Translates a tag-like bound (such as `1.62.0`) to a `Bound::Date` so that
+/// bisecting works for versions older than 167 days.
+fn translate_tags(args: &Opts) -> anyhow::Result<(Option<Bound>, Option<Bound>)> {
+    let is_tag = |bound: &Option<Bound>| -> bool {
+        match bound {
+            Some(Bound::Commit(commit)) => commit.contains('.'),
+            None | Some(Bound::Date(_)) => false,
+        }
+    };
+    let is_datelike = |bound: &Option<Bound>| -> bool {
+        matches!(bound, None | Some(Bound::Date(_))) || is_tag(bound)
+    };
+    if !(is_datelike(&args.start) && is_datelike(&args.end)) {
+        // If the user specified an actual commit for one bound, then don't
+        // even try to convert the other bound to a date.
+        return Ok((args.start.clone(), args.end.clone()));
+    }
+    let fixup = |which: &str, bound: &Option<Bound>| -> anyhow::Result<Option<Bound>> {
+        if is_tag(bound) {
+            if let Some(Bound::Commit(tag)) = bound {
+                let date = args
+                    .access
+                    .repo()
+                    .bound_to_date(Bound::Commit(tag.clone()))?;
+                eprintln!(
+                    "translating --{which}={tag} to {date}",
+                    date = date.format(YYYY_MM_DD)
+                );
+                return Ok(Some(Bound::Date(date)));
+            }
+        }
+        Ok(bound.clone())
+    };
+    Ok((fixup("start", &args.start)?, fixup("end", &args.end)?))
+}
+
+/// Returns the commit SHA of the nightly associated with the given date.
+fn date_to_sha(date: &NaiveDate) -> anyhow::Result<String> {
+    let date_str = date.format(YYYY_MM_DD);
+    let url = format!("{NIGHTLY_SERVER}/{date_str}/channel-rust-nightly-git-commit-hash.txt");
+
+    eprintln!("fetching {url}");
+    let client = Client::new();
+    let name = format!("nightly manifest {date_str}");
+    let mut response = download_progress(&client, &name, &url)?;
+    let mut commit = String::new();
+    response.read_to_string(&mut commit)?;
+
+    eprintln!("converted {date_str} to {commit}");
+
+    Ok(commit)
+}
+
+/// Returns the date of the nightly toolchain currently installed. If no
+/// nightly is found, then it goes to the network to determine the date of the
+/// latest nightly.
+fn installed_nightly_or_latest() -> anyhow::Result<GitDate> {
+    if let Some(date) = Toolchain::default_nightly() {
+        return Ok(date);
+    }
+    find_latest_nightly()
+}
+
+/// Returns the date of the latest nightly (fetched from the network).
+fn find_latest_nightly() -> anyhow::Result<GitDate> {
+    let url = format!("{NIGHTLY_SERVER}/channel-rust-nightly.toml");
```

Comment:
https://static.rust-lang.org/dist/channel-rust-nightly-date.txt maybe works? Not sure if it's uploaded late enough in the process...

Reply:
Oh, that's better! I had forgotten about that.

From what I can tell, it looks like it should work, so I went ahead and switched to the txt file. From what I can see in promote-release, the [last step](https://github.com/rust-lang/promote-release/blob/dd7a74323e1a3b489d452620732574098afefc76/src/main.rs#L248) it takes involves copying the manifests, including this txt file, and that happens after the archives have been uploaded.